src/Engine/Component/AnimationController.cpp

```diff
@@ -28,19 +30,66 @@ void Component::AnimationController::UpdateAnimation(float deltaTime) {
     if (skeleton == nullptr || controller == nullptr)
         return;
 
-    if (bones.size() != skeleton->skeletonBones.size()) {
-        bones.clear();
-        bones.shrink_to_fit();
-        bones.resize(skeleton->skeletonBones.size());
+    if (skeleton != nullptr) {
+        if (bones.size() != skeleton->skeletonBones.size()) {
+            bones.clear();
+            bones.shrink_to_fit();
```

Comment:
No need to shrink only to resize it immediately after.

Reply:
Didn't get the correct size when did that, maybe because it wont resize to something smaller? However I will try without it, size it's pretty expensive.

Reply:
Yea, I'm fairly certain resize does nothing if capacity is already sufficient.